maths/allocation_number.nim

```diff
@@ -0,0 +1,52 @@
+## In a multi-threaded download, this algorithmm could be used to provide
```

Comment:
I could not find any reference to any "named algorithm", the only thing I see is indeed the algorithm in python (which has a different api since it returns a string with the range): https://github.com/TheAlgorithms/Python/blob/master/maths/allocation_number.py

honestly I would probably skip this, but in case you want to keep it:
- you might want to add a test with where numPartitions does not divide exactly numBytes
- you would need to pick a name of algorithm to have as first line in a documentation comment if we plan to merge this: https://github.com/TheAlgorithms/Nim/pull/5

Reply:
I removed this « algorithm ». It was not clear what it was about exactly.